### PR TITLE
Enhancement/6247 most popular content by pageviews

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue:

- #6247 

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [ ] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics-4/components/widgets/PopularContentWidget.js [hunk 1]
```diff
@@ -21,31 +21,101 @@
...
 
-export default function PopularContentWidget( { Widget, WidgetNull } ) {
-	const keyMetricsWidgetHidden = useSelect( ( select ) =>
-		select( CORE_USER ).isKeyMetricsWidgetHidden()
+	const report = useInViewSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getReport( reportOptions )
 	);
 
-	if ( keyMetricsWidgetHidden !== false ) {
-		return <WidgetNull />;
```
Comment (techanvil) [inline assets/js/modules/analytics-4/components/widgets/PopularContentWidget.js:38]: Hmm, I notice this widget now doesn't return null, but the IB does specify that it _should_ return null under a couple of circumstances:

  * Return `null` if the `GA4` module is not connected.
  * Using the `getKeyMetrics` selector, check if any of the selected Metric is from GA4. If not, return `null`. Otherwise, continue.

Now, I have also spotted that the QAB says "For now, all widgets are displayed on the dashboard (which will change with future commits)", however the only relevant issue I can see is https://github.com/google/site-kit-wp/issues/7061, which should take care of the second point relating to `getKeyMetrics`; the point about returning `null` (actually, `WidgetNull` would be more appropriate) if GA4 is not connected still seems like it's probably relevant. We are already doing something similar in at least one of the other Key Metrics widgets, the `ConnectGA4CTATileWidget`:

https://github.com/google/site-kit-wp/blob/d7bfa9887ff7e6f908c1d6c3f173dc68b22bd975/assets/js/modules/analytics-4/components/widgets/ConnectGA4CTATileWidget.js#L62-L64

So, to summarise, it looks like returning `WidgetNull` when GA4 is not connected is probably still the right thing to do here, whereas the point about checking the `getKeyMetrics` selector will effectively be taken care of later via #7061. What do you think?

File: tests/backstop/reference/google-site-kit_KeyMetrics_MetricTileTable_Ready_0_document_0_small.png
Comment (techanvil) [inline tests/backstop/reference/google-site-kit_KeyMetrics_MetricTileTable_Ready_0_document_0_small.png:1]: It could be worth making at least one of the fields here quite long so it demonstrates the `text-overflow: ellipses` at a narrow breakpoint...

File: assets/js/components/KeyMetrics/MetricTileTable.js [hunk 2]
```diff
@@ -90,9 +97,9 @@ export default function MetricTileTable( props ) {
 						/>
 					) }
 					{ ! loading && (
-						<table className="googlesitekit-km-widget-tile__table">
-							<tbody>{ tbody }</tbody>
-						</table>
+						<div className="googlesitekit-km-widget-tile__table">
+							{ tbody }
```
Comment (techanvil) [inline assets/js/components/KeyMetrics/MetricTileTable.js:101]: As the markup is no longer using a `table`, this `tbody` name seems a little out of place. I'd suggest renaming it, how about `tileBody`?

```suggestion
							{ tileBody }
```

**Review by techanvil — CHANGES_REQUESTED** [review]
This is looking good @eugene-manuilov. I've left a handful of comments, please take a look.

File: assets/js/modules/analytics-4/components/widgets/PopularContentWidget.js [hunk 3]
```diff
@@ -21,31 +21,101 @@
...
+	const { rows = [] } = report || {};
+
+	const columns = [
+		{
+			field: 'dimensionValues',
+			Component: ( { fieldValue } ) => {
+				const [ title, url ] = fieldValue;
+
+				return (
+					<Link href={ url.value } title={ title.value }>
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply assets/js/modules/analytics-4/components/widgets/PopularContentWidget.js]: Good point. Updated.

File: assets/js/components/KeyMetrics/MetricTileTable.js (hunk 2 above)
Comment (eugene-manuilov) [reply assets/js/components/KeyMetrics/MetricTileTable.js]: Updated.

File: tests/backstop/reference/google-site-kit_KeyMetrics_MetricTileTable_Ready_0_document_0_small.png
Comment (eugene-manuilov) [reply tests/backstop/reference/google-site-kit_KeyMetrics_MetricTileTable_Ready_0_document_0_small.png:1]: Yes, makes sense. Updated.

**Review by techanvil — APPROVED** [review]
LGTM, nice one @eugene-manuilov!
